d2bs/kolbot/libs/common/Prototypes.js

```diff
@@ -1104,3 +1104,149 @@ if (!Array.prototype.find) {
 		writable: true
 	});
 }
+
+/**
+ * @description very simple but handy function, that either returns the first element, or undefined
+ */
+if (!Array.prototype.first) {
+	Array.prototype.first = function () {
+		return this.length > 0 ? this[0] : undefined;
+	};
+}
+
+/**
+ * Just a simple wrapper around unit.getItem(), to avoid the use of .getNext() but just get an array,
+ * and the guaranty of getting an array ack if the unit doesnt have items
+ * @param args
+ * @returns Unit[]
+ */
+Unit.prototype.getItems = function (...args) {
+	let item = this.getItem.apply(this, args), items = [];
+	item && items.push(copyUnit(item));
+
+	while (item && item.getNext()) {
```

Comment:
change to do while loop to prevent the pre push before the loop

do {
    if (item) {
        items.push(copyUnit(item));
    }
} while (item.getNext())